fork-off/src/main.rs

```diff
@@ -56,9 +56,12 @@ async fn main() -> anyhow::Result<()> {
     }
     let state = read_snapshot_from_file(snapshot_path);
 
-    let initial_state: Storage =
-        serde_json::from_value(initial_spec["genesis"]["raw"]["top"].take())
-            .expect("Deserialization of state from given chainspec file failed");
+    // Initialize with state from chainspec + empty child storage
+    let initial_state = Storage {
+        top: serde_json::from_value(initial_spec["genesis"]["raw"]["top"].take())
+            .expect("Deserialization of state from given chainspec file failed"),
+        child_storage: ChildStorage::new(),
+    };
```

Comment:
We can create a constructor for `Storage` that accepts the `initial_spec` only:
```rust
impl Storage {
  pub fn new(initial_spec) -> Result<Self, anyhow::Error> {
    ...
  }
}
```

Reply:
done